diff --git a/Model/AccountManagement.php b/Model/AccountManagement.php
index 9173307a7d27..8beecffd1c86 100644
--- a/Model/AccountManagement.php
+++ b/Model/AccountManagement.php
@@ -670,8 +670,8 @@
         $customerSecure->setRpTokenCreatedAt(null);
         $customerSecure->setPasswordHash($this->createPasswordHash($newPassword));
         $this->getAuthentication()->unlock($customer->getId());
-        $this->sessionManager->destroy();
         $this->destroyCustomerSessions($customer->getId());
+        $this->sessionManager->destroy();
         $this->customerRepository->save($customer);
 
         return true;
